python/python/bystro/utils/tar.py

```diff
@@ -0,0 +1,77 @@
+"""Utility functions for safely invoking tar in cross-OS manner."""
+
+import subprocess
+from enum import Enum
+from typing import NoReturn
+
+
+class OperatingSystem(Enum):
+    """Represent the operating system on which the current Python process is running."""
+
+    linux = "linux"
+    macosx = "macosx"
+
+
+# here and throughout, fully qualify executable names to avoid privilege escalation
+GNU_TAR_LINUX = "/usr/bin/tar"
+GNU_TAR_MACOSX = "/opt/homebrew/bin/gtar"
+
+
+def assert_never(value: NoReturn) -> NoReturn:
+    """Raise error for mypy if code is proven reachable."""
+    msg = f"Unhandled value: {value} ({type(value).__name__})"
+    raise AssertionError(msg)
+
+
+def _determine_os() -> OperatingSystem:
+    """Determine whether we're running on Linux or MacOSX by inspecting uname output."""
+    try:
+        result = subprocess.run(
+            ["/usr/bin/uname", "-a"],  # noqa: S603 (this input is safe)
+            capture_output=True,
+            text=True,
+            check=True,
+        )
+    except FileNotFoundError as file_not_found_error:
+        err_msg = (
+            "Couldn't run `uname -a` in shell to determine OS: only linux and macosx are supported."
+        )
+        raise AssertionError(err_msg) from file_not_found_error
+    downcased_uname_output = result.stdout.lower()
+    if "darwin" in downcased_uname_output:
+        return OperatingSystem.macosx
+    if "linux" in downcased_uname_output:
+        return OperatingSystem.linux
+    err_msg = f"Could not determine OS from `uname -a` output: `{result.stdout}`"
+    raise OSError(err_msg)
+
+
+def _get_gnu_tar_executable_name() -> str:
+    """Find the name of the GNU tar executable on user's instance."""
+    # Macs use bsdtar by default.  bsdtar is not fully compatible with
+    # GNU tar, the implementation we use.  So if we're on a mac, we'll
+    # want to use 'gtar' instead of 'tar'.
```

Comment:
This makes sense; could we update the Mac install script to install gtar? It also strikes me that we'll need to do this in Perl, which we'll need an issue for.